Add tests for git metrics

JonathanHallam opened the pull request.

JonathanHallam committed:
Add tests for git metrics

kentsanggds commented on tests/metrics/test_github.py:
```diff
@@ -0,0 +1,169 @@
+
+def test_get_git_metrics(mocker):
+    mock_github_client(mocker)
+    dao_mock = mocker.patch("app.source.github.dao_upsert_git_metric", return_value={"this_is": "a mock response"})
```
```suggestion
    dao_mock = mocker.patch("app.source.github.dao_upsert_git_metric")
```

kentsanggds commented on tests/metrics/test_github.py:
```diff
@@ -0,0 +1,169 @@
+            self.token = token
+
+        def search_repositories(self, query):
+            return [Repository(no_prs, create_date)]
```
Don't need to pass in `no_prs` or `create_date` from the bottom, it can just be referred to in the function that needs it as it is should still be in scope.

kentsanggds commented on tests/metrics/test_github.py:
```diff
@@ -0,0 +1,169 @@
+
+def test_get_git_metrics_only_in_quarter(mocker):
+    mock_github_client(mocker)
+    dao_mock = mocker.patch("app.source.github.dao_upsert_git_metric", return_value={"this_is": "a mock response"})
```
```suggestion
    dao_mock = mocker.patch("app.source.github.dao_upsert_git_metric")
```

kentsanggds commented on tests/metrics/test_github.py:
```diff
@@ -0,0 +1,169 @@
+def test_get_git_metrics_includes_prs_started_before_quarter(mocker):
+    date = datetime.datetime(2018, 12, 8)
+    mock_github_client(mocker, create_date=date)
+    dao_mock = mocker.patch("app.source.github.dao_upsert_git_metric", return_value={"this_is": "a mock response"})
```
```suggestion
    dao_mock = mocker.patch("app.source.github.dao_upsert_git_metric")
```

kentsanggds commented on tests/metrics/test_github.py:
```diff
@@ -0,0 +1,169 @@
+
+def test_get_git_metrics_ignores_repos_without_prs(mocker):
+    mock_github_client(mocker, no_prs=True)
+    dao_mock = mocker.patch("app.source.github.dao_upsert_git_metric", return_value={"this_is": "a mock response"})
```
```suggestion
    dao_mock = mocker.patch("app.source.github.dao_upsert_git_metric")
```

kentsanggds commented on tests/metrics/test_github.py:
```diff
@@ -0,0 +1,168 @@
+
+    Github().get_metrics(year=2018, quarter=2)
+    dao_mock.assert_not_called()
+
```
2 blank lines between functions not in a class required for pycodestyle.

kentsanggds commented on tests/metrics/test_github.py:
```diff
@@ -0,0 +1,168 @@
+        'diff_count': 120,
+        'total_diff_count': 120,
+        'comments_count': 4})
+
```
2 blank lines between functions not in a class required for pycodestyle.

kentsanggds commented on tests/metrics/test_github.py:
```diff
@@ -0,0 +1,168 @@
+            return RepoCommit()
+
+    class RepoCommit:
+        files = [{
```
Looks like the mock data is hard coded, would be better if the mock data can be passed in to test different scenarios.

kentsanggds committed:
Update code for pycodestyle format

kentsanggds committed:
wip - update test github to create better test framework

Have done a rough update of pull reqs but could be refined.

Commits should follow in same line as pull reqs.

JonathanHallam committed:
change test itteration behaviour

kentsanggds commented on tests/metrics/test_github.py:
```diff
@@ -0,0 +1,214 @@
+    commits = []
+
+    for date in commit_dates_arr:
+        commits.append(Commit(**date))
```
You shouldn't have to unpack the dates

JonathanHallam replied:
sorted

kentsanggds reviewed (changes requested):
Just a small change, I don't think it's necessary to unpack dates when creating `Commit` object, it should be ok to pass in the date.

kentsanggds commented on tests/metrics/test_github.py:
```diff
@@ -0,0 +1,225 @@
+
+    class PullReq:
+        comments_count = 0
+        review_comments_count = 4
```
Would be good if we can also move the `comments_count` and `review_comments_count` into the `pr_dict`

kentsanggds reviewed (changes requested):
Also, I've noticed that `test_github.py` is still in `metrics` folder, can it be moved to `source` folder for consistency.
